src/MatrixTypes_Native.f90

```diff
@@ -801,5 +813,40 @@ SUBROUTINE transpose_SparseMatrixType(matrix)
       DEALLOCATE(n_Row)
       DEALLOCATE(Row)
     ENDSUBROUTINE transpose_SparseMatrixType
-    
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_SparseMatrixType(matrix)
+      CLASS(SparseMatrixType),INTENT(INOUT) :: matrix
+    ENDSUBROUTINE zeroentries_SparseMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_DenseSquareMatrixType(matrix)
+      CLASS(DenseSquareMatrixType),INTENT(INOUT) :: matrix
+    ENDSUBROUTINE zeroentries_DenseSquareMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_TriDiagMatrixType(matrix)
+      CLASS(TriDiagMatrixType),INTENT(INOUT) :: matrix
+    ENDSUBROUTINE zeroentries_TriDiagMatrixType
+!
+!-------------------------------------------------------------------------------
+!> @brief zero the matrix
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE  zeroentries_DenseRectMatrixType(matrix)
+      CLASS(DenseRectMatrixType),INTENT(INOUT) :: matrix
```

Comment:
Implement and test

Reply:
I don't believe for now there's a need for implementing this. Since the problem is only with PETSC.

Reply:
They should all be implemented and tested so there's consistent behavior for clients of the package. They may want to switch matrix types, and if they do, their code won't work, and it will not be obvious.

Implementing and testing these is really important to the overall quality of the package.